int32 v1_min
int32 v2_min
int32 v3_min
int32 v1_max
int32 v2_max
int32 v3_max
